[UnifiedPDF] Describe selection rects in terms of PDFPageCoverage

smfr opened the pull request:
#### 5d5ae59db1cbc2bcc2fd74d11dce31f9a8c86ef8
<pre>
[UnifiedPDF] Describe selection rects in terms of PDFPageCoverage
<a href="https://bugs.webkit.org/show_bug.cgi?id=274139">https://bugs.webkit.org/show_bug.cgi?id=274139</a>
<a href="https://rdar.apple.com/128043619">rdar://128043619</a>

Reviewed by Abrar Rahman Protyasha.

Replace `UnifiedPDFPlugin::boundsForSelection()` with `UnifiedPDFPlugin::pageCoverageForSelection()`
which returns a `PDFPageCoverage`, which will be useful in future when we need to have per-page
selection bounds info.

`UnifiedPDFPlugin::textIndicatorForSelection()` can use this, asking for only the first page.
`UnifiedPDFPlugin::selectionBoundsForFirstPageInDocumentSpace()` can be removed.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::pageCoverageForSelection const):
(WebKit::UnifiedPDFPlugin::boundsForSelection const):
(WebKit::UnifiedPDFPlugin::repaintOnSelectionActiveStateChangeIfNeeded):
(WebKit::UnifiedPDFPlugin::setCurrentSelection):
(WebKit::UnifiedPDFPlugin::textIndicatorForSelection):
(WebKit::UnifiedPDFPlugin::selectionBoundsForFirstPageInDocumentSpace const): Deleted.

Canonical link: <a href="https://commits.webkit.org/278759@main">https://commits.webkit.org/278759@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/389e50750d3c7354665c09ab2357789024f85c4a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51445 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30755 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3796 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54712 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2138 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37087 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1818 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41890 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53544 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28402 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44366 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23015 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25708 "Passed tests") | | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47677 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1706 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56304 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26564 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1585 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49282 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27804 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44427 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48465 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11265 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28697 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27539 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -3284,17 +3284,39 @@ static FloatRect computeMarqueeSelectionRect(const WebCore::FloatPoint& point1,
+
+Vector<FloatRect> UnifiedPDFPlugin::boundsForSelection(PDFSelection *selection, CoordinateSpace targetSpace) const
+{
+    auto pageCoverage = pageCoverageForSelection(selection);
+    if (pageCoverage.isEmpty())
+        return { };
+
+    Vector<FloatRect> pageRects;
+    pageRects.reserveInitialCapacity(pageCoverage.size());
+
+    for (auto& page : pageCoverage)
+        pageRects.append(convertUp(CoordinateSpace::PDFPage, targetSpace, page.pageBounds, page.pageIndex));
+
+    return pageRects;
```
nit: You could use `Vector::map` for this one? The `map` implementation is basically what you have here!

```suggestion
    return pageCoverageForSelection(selection).map([this, targetSpace](auto& page) {
        return convertUp(CoordinateSpace::PDFPage, targetSpace, page.pageBounds, page.pageIndex);
    });
```

smfr committed:
[UnifiedPDF] Describe selection rects in terms of PDFPageCoverage
https://bugs.webkit.org/show_bug.cgi?id=274139
rdar://128043619

Reviewed by Abrar Rahman Protyasha.

Replace `UnifiedPDFPlugin::boundsForSelection()` with `UnifiedPDFPlugin::pageCoverageForSelection()`
which returns a `PDFPageCoverage`, which will be useful in future when we need to have per-page
selection bounds info.

`UnifiedPDFPlugin::textIndicatorForSelection()` can use this, asking for only the first page.
`UnifiedPDFPlugin::selectionBoundsForFirstPageInDocumentSpace()` can be removed.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::pageCoverageForSelection const):
(WebKit::UnifiedPDFPlugin::boundsForSelection const):
(WebKit::UnifiedPDFPlugin::repaintOnSelectionActiveStateChangeIfNeeded):
(WebKit::UnifiedPDFPlugin::setCurrentSelection):
(WebKit::UnifiedPDFPlugin::textIndicatorForSelection):
(WebKit::UnifiedPDFPlugin::selectionBoundsForFirstPageInDocumentSpace const): Deleted.

Canonical link: https://commits.webkit.org/278759@main
